docker_101/docker_101.md

```diff
@@ -0,0 +1,427 @@
+
+The Dockerfile is a set of instructions for generating a container image. You can think of it like a recipe: It tells the computer what ingredients to use and how to combine them to create the container.
+
+A text description about the project sounds more like a [README file](https://www.makeareadme.com/) --- a great thing to include in your project, just not in your Dockerfile. 
+
+Note that you can (and should!) include comments in your Dockerfile describing what it does and why. This makes your Dockerfile much more useful to anyone trying to replicate your work, including future you looking back at your old code. 
+
+</div>
+***
+
+In the command `docker build -t getting-started .` what does the `-t` flag stand for?
+
+[[tag]]
+<script>
+  let input = "@input".trim().toLowerCase();
+  input == "tag";
+</script>
+***
+<div class = "answer">
+
+It refers to a tag, a custom identifier you can use to refer to this image (instead of relying on the long, random IDs Docker assigns by default). 
+The tag in this example is `getting-started`, so if we want to run this container we just built we can use a command like `docker run getting-started` and Docker will know we're talking about the image we built and tagged above.
+
+To learn more, see the [Docker handbook section on tags](https://docker-handbook.farhan.dev/image-manipulation-basics#tagging-images).
+
+</div>
+***
+
+True or False: The command to create a container image is `docker build`.
+
+[(X)] True
+[( )] False
+***
+<div class = "answer">
+
+Yes, you use `docker build` to create a container image.
+The command `docker run` starts a container that you've already built. 
+
+</div>
+***
+
+Once you've built a Docker image, what is the easiest way to share it?
+
+[( )] Send the Dockerfile via email
+[(X)] Share the image via an online repository
+[( )] Either of the above will work
+***
+<div class = "answer">
+
+The best way to share a container image is with the [Docker Hub](https://hub.docker.com/) repository. For a review, see the "Sharing our App" section of the tutorial.
+
+Note that sending just the Dockerfile to someone isn't enough for them to be able to build the container image (just as sending a recipe to someone won't let them cook the dish unless you also send the ingredients), so that won't work. 
+
+</div>
+***
+
+## Using Docker in Research
+
+Most of the tutorials and explanations you'll find online about Docker (including the Docker 101 tutorial you just worked on) focus on its use in software development. 
+Importantly, most of the example Dockerfiles you'll find online are also written by and for developers. 
+As someone new to Docker (and even for experienced users!), being able to copy existing Dockerfiles is an important time saver.
+Although you might be able to get these images to work for your research needs, there are a lot of special considerations that come up when you're using containers for reproducible research rather than software develpoment. 
```

Comment:
```suggestion
Although you might be able to get these images to work for your research needs, there are a lot of special considerations that come up when you're using containers for reproducible research rather than software development. 
```